### PR TITLE
ExprYawPitch - fix issue with not being able to set yaw/pitch of location variables

### PR DESCRIPTION
**Pull request by ShaneBeee**
### Description
This PR aims to fix an issue where setting the yaw/pitch of a location variable did not work. 
Snow suggested merging the two yaw/pitch expressions (location and vector) together

I have copied the vector class into this one. This is a preliminary run at the situation. 
If y'all want anything changed, lemme know.

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/expressions/ExprYawPitch.java [hunk 1]
```diff
@@ -31,90 +32,133 @@
...
+			case ADD:
+				if (usesYaw)
+					yaw += n;
+				else
+					pitch -= n; // Negative because of Minecraft's / Skript's upside down pitch
+				vector = VectorMath.fromYawAndPitch(yaw, pitch);
+				getExpr().change(e, new org.bukkit.util.Vector[]{vector}, ChangeMode.SET);
+				break;
+			case REMOVE:
+				n = -n;
```
Comment (Nicofisi) [inline src/main/java/ch/njol/skript/expressions/ExprYawPitch.java:134]: Is this REMOVE supposed to use the reverse SET logic and not the reverse ADD logic?

**Review by Nicofisi — COMMENTED** [review]
Looks good in general!

File: src/main/java/ch/njol/skript/expressions/ExprYawPitch.java (hunk 1 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprYawPitch.java:134]: I didnt write this stuff, just copied from the Vector Yaw/Pitch class, so I have no clue to be honest.

Comment (APickledWalrus) [reply src/main/java/ch/njol/skript/expressions/ExprYawPitch.java:134]: yeah that seems odd to me... maybe REMOVE should go above ADD?

File: src/main/java/ch/njol/skript/expressions/ExprYawPitch.java [hunk 2]
```diff
@@ -31,90 +32,133 @@
...
-	}
-	
-	@SuppressWarnings({"null"})
-		@Override
-		public Class<?>[] acceptChange(final ChangeMode mode) {
-			if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE)
-				return CollectionUtils.array(Number.class);
-			return null;
+	public void change(Event e, Object[] delta, ChangeMode mode) {
+		if (delta[0] == null)
```
[the remark this answers is not in the document]
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprYawPitch.java]: ya fair enough
fixed

**Review by APickledWalrus — APPROVED** [review]

**Review by APickledWalrus — APPROVED** [review]
image
I forgot your seal of approval!

**Pull request merged**